### ISSUE TITLE
useBuiltins: 'usage' fails with babel-loader when not excluding node_modules

### ISSUE THREAD

**Issue by philipwalton** [babel/babel#7559]
Choose one: is this a bug report or feature request?

Bug

### Input Code

After trying out the new `useBuiltins: 'usage'` feature I was getting the following error:

```
bundle.js:181 Uncaught TypeError: __webpack_require__(...) is not a function
```

Line 180-182 looks like this (they're from the core-js polyfills):

```js
// @@split logic
__webpack_require__(14)('split', 2, function (defined, SPLIT, $split) {
  'use strict';
```

I'm not 100% why this function isn't defined, but I discovered it only happens when I don't exclude `/node_modules/` in my `babel-loader` config.

Here's the reduce code and webpack config you need to reproduce the error:

**index.js:**

```js
console.log('foo'.replace(/o/g, '0'));
```

**webpack.config.js:**

```js
module.exports = {
  entry: './index.js',
  output: {
    filename: 'bundle.js',
  },
  module: {
    rules: [{
      test: /\.js$/,
      use: {
        loader: 'babel-loader',
        options: {
          babelrc: false,
          // exclude: /node_modules/,
          presets: [
            ['@babel/env', {
              debug: true,
              modules: false,
              useBuiltIns: 'usage',
            }],
          ],
        },
      },
    }],
  },
};
```

### Babel/Babylon Configuration (.babelrc, package.json, cli command)

```js
{
  "devDependencies": {
    "@babel/core": "^7.0.0-beta.40",
    "@babel/preset-env": "^7.0.0-beta.40",
    "babel-loader": "^8.0.0-beta.0",
    "webpack": "^3.11.0"
  }
}
```

### Expected Behavior

The `useBuiltins: 'usage'` feature should support loading files in `node_modules`. I.e. it should be able to handle loading `core-js` (which it loads itself).

### Current Behavior

Webpack bundles without error, but the file errors when run in the browser.

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            |  7.0.0-beta.40
| Babylon          | 
| node             |  9.7.1
| npm              | 5.6.1
| Operating System | Mac OS X 10.13.3

**Comment by Kovensky** [comment]
The plugin is _also_ processing the core-js source code, so it's trying to use core-js to polyfill the things core-js is trying to polyfill 😆

Adding `/\bcore-js\b/` to `excludes` should fix that particular problem, but it still won't be usable for transforming `node_modules` until https://github.com/babel/babel/issues/7457 is fixed.

**Comment by philipwalton** [comment]
Thanks, I hadn't seen #7457. I'll keep an eye on that issue as well.

**Comment by Kovensky** [comment]
This should be solved now; you still have to exclude `/\bcore-js\b/` from `babel-loader` to avoid that circular dependency loop; otherwise just make sure to set `sourceType: 'unambiguous'` in your babel config.

**Comment by bodograumann** [comment]
I just came across this issue myself. It was a real pain until I found this thread.
It would probably be best to add some hint in the documentation of `useBuiltIns: 'usage'`.
The following solutions worked for me:

- Adding **both** `sourceType: 'unambiguous'` in *babel.config.js* and `exclude: /\bcore-js\b/` in the *webpack.config.js*.
  That is probably what you said, @Kovensky, but it wasn’t completely clear to me, until I tried it.
  I’m still not sure, what `sourceType` does in this context.
- Just adding `exclude: /\bnode-modules\b/` in *webpack.config.js*.

I went with the first one, because it will allow me to transpile other code in *node_modules* in the future.
